zerver/lib/outgoing_webhook.py

```diff
@@ -0,0 +1,72 @@
+from __future__ import absolute_import
+from typing import Any, Iterable, Dict, Optional, Set, Tuple
+from six import text_type
+
+import requests
+import sys
+import inspect
+from six.moves import urllib
+
+from zerver.models import get_outgoing_webhook_bot_profile, Realm
+
+from straight.plugin import load
+
+class OutgoingWebhookBotInterface(object):
+    email = None # type: text_type
+    full_name = None # type: text_type
+
+    def __init__(self):
+        self.post_url = None # type: text_type
+        self.service_api_key = None # type: text_type
+
+    def process_command(self, command):
+        raise NotImplementedError()
+
+    def process_response(self, response_json):
+        raise NotImplementedError()
+
+    def handle_failure(self, status_code, json_response):
+        raise NotImplementedError()
+
+class OutgoingWebhookBot():
```

Comment:
Ah yes. I wanted something like a dynamic inheritance but I was not sure I could do it in python. It seems you can totally do something like that!